### ISSUE TITLE
More info in callbacks

### ISSUE THREAD

**Issue by antoine-levitt** [JuliaNLSolvers/Optim.jl#857]
Just ran into this the other day, I wanted to access the current iterate in the callback and couldn't. In general I'm a big fan of passing as much information as available in callbacks (possibly not documenting it so you don't commit to an API). I also find named tuples are nice for this, instead of typed structs or dicts.

**Comment by pkofod** [comment]
Yup, I totally agree. I think there has been a discussion around this before. Do you have any specific ideas? This will be very breaking and I was thinking of tagging tonight :)

**Comment by antoine-levitt** [comment]
Nothing special, I worked around it, just thought I'd open the issue as a feature request for your fabled new rewrite :-p

**Comment by pkofod** [comment]
Okay. I actually haven't included callbcks yet but it's a very good point

**Comment by pkofod** [comment]
Done in #1224
